.github/container/test-t5x-fmha.sh

```diff
@@ -0,0 +1,244 @@
+#!/bin/bash
```

Comment:
Can the hlo dump feature be merged with test-t5x.sh since saving hlo dumps sounds generally useful?

Here's the current diff:
```diff
$ diff test-t5x*
23d22
<     echo "  --save-hlo {0, 1}         1 to save the dumped hlo, 0 to remove the hlo dumped folder"
47d45
< SAVE_HLO=${SAVE_HLO:-1}
87,90d84
<         --save-hlo)
<             SAVE_HLO="$2"
<             shift 2
<             ;;
114,119d107
< # Set hlo dump folder after output folder is set.
< HLO_DIR=${OUTPUT}/hlo
<
< ## Setting the env variables for FMHA
< export XLA_FLAGS="--xla_dump_hlo_as_text --xla_dump_to=${HLO_DIR} --xla_gpu_graph_level=0 --xla_gpu_fused_attention_use_cudnn_rng=true"
<
134,135d121
< print_var HLO_DIR
< print_var SAVE_HLO
223,244d208
<
< ## Check if fmha instructions are present in the HLO dumped file or not.
< fmha_regex="fmha[-bmm]?[-scale]?[-bias]?[-mask]?[-softmax]?[-dropout]?[-bmm]?[-backward]?*"
<
< result=$(grep -irlnE "$fmha_regex" "${HLO_DIR}/"*.txt)
<
< if [[ $SAVE_HLO -eq 0 ]]; then
<       rm -rf $HLO_DIR
<       echo "Removed dumped HLO directory!"
< fi
<
< if [ -z "$result" ]; then
<         echo "E: No FMHA instructions were found in the hlo files!"
<       exit 1
< else
<       echo -e "Found FMHA instructions in the following HLO files: \n $result" | while read -r line; do
<               echo "$line"
<       done
<         exit 0
<
< fi
<
```
The hlo dump gets put into the OUTPUT_DIR. How large is the hlo dump for this model size (since it will be saved as a github artifact)? 

At the moment, I feel that this whole script should be merged with test-t5x.sh and the "check fmha in hlo dump" be behind a flag so that it's not tested if not enabled. What do you think?

Reply:
Merged and enabled dumping of hlo folder in t5x as well. The XLA dev team also wants to dump the hlo for paxml tests, will add changes for that as part of FMHA for paxml. The hlo folder size is 49-55 MB per testcase job.